File: surface/widgets/appkit/components/widgets/navigation/Breadcrumb.swift
// ============================================================
// Clef Surface AppKit Widget — Breadcrumb
//
// Horizontal navigation trail showing the current location
// hierarchy. Each crumb is clickable except the last.
// ============================================================

import AppKit

public class ClefBreadcrumbView: NSView {
    public struct Crumb {
        public let label: String
        public let action: (() -> Void)?
        public init(label: String, action: (() -> Void)? = nil) {
            self.label = label
            self.action = action
        }
    }

    public var crumbs: [Crumb] = [] { didSet { rebuild() } }
    public var separator: String = "/"

    private let stackView = NSStackView()

    public override init(frame frameRect: NSRect) {
        super.init(frame: frameRect)
        setup()
    }

    required init?(coder: NSCoder) {
        super.init(coder: coder)
        setup()
    }

    private func setup() {
        stackView.orientation = .horizontal
        stackView.spacing = 4
        addSubview(stackView)
    }

    private func rebuild() {
        stackView.arrangedSubviews.forEach { stackView.removeArrangedSubview($0); $0.removeFromSuperview() }
        for (i, crumb) in crumbs.enumerated() {
            if i > 0 {
                let sep = NSTextField(labelWithString: separator)
                sep.font = NSFont.systemFont(ofSize: 12)
                sep.textColor = .tertiaryLabelColor
                stackView.addArrangedSubview(sep)
            }
            let isLast = i == crumbs.count - 1
            if isLast {
                let label = NSTextField(labelWithString: crumb.label)
                label.font = NSFont.systemFont(ofSize: 12, weight: .medium)
                label.textColor = .labelColor
                label.isEditable = false
                label.isBezeled = false
                label.drawsBackground = false
                stackView.addArrangedSubview(label)
            } else {
                let btn = NSButton(title: crumb.label, target: self, action: #selector(crumbClicked(_:)))
                btn.tag = i
                btn.bezelStyle = .inline
                btn.isBordered = false
                btn.font = NSFont.systemFont(ofSize: 12)
                btn.contentTintColor = .controlAccentColor
                stackView.addArrangedSubview(btn)
            }
        }
    }

    @objc private func crumbClicked(_ sender: NSButton) {
        guard sender.tag < crumbs.count else { return }
        crumbs[sender.tag].action?()
    }

    public override func layout() {
        super.layout()
        stackView.frame = bounds
    }
}

File: surface/widgets/nativescript/components/widgets/navigation/Breadcrumb.ts
// ============================================================
// Clef Surface NativeScript Widget — Breadcrumb
//
// Navigation breadcrumb trail showing page hierarchy.
// ============================================================

import { StackLayout, Label } from '@nativescript/core';

export interface BreadcrumbItem { label: string; href?: string; }

export interface BreadcrumbProps {
  items: BreadcrumbItem[];
  separator?: string;
  onNavigate?: (href: string) => void;
}

export function createBreadcrumb(props: BreadcrumbProps): StackLayout {
  const { items, separator = '/', onNavigate } = props;
  const container = new StackLayout();
  container.className = 'clef-widget-breadcrumb';
  container.orientation = 'horizontal';
  container.accessibilityRole = 'none';

  items.forEach((item, index) => {
    const link = new Label();
    link.text = item.label;
    link.className = index === items.length - 1 ? 'clef-breadcrumb-current' : 'clef-breadcrumb-link';
    if (item.href && index < items.length - 1) {
      link.on('tap', () => onNavigate?.(item.href!));
    }
    container.addChild(link);
    if (index < items.length - 1) {
      const sep = new Label();
      sep.text = ` ${separator} `;
      sep.opacity = 0.5;
      container.addChild(sep);
    }
  });
  return container;
}

export default createBreadcrumb;
